include/deal.II/distributed/tria.h

```diff
@@ -880,7 +881,37 @@ namespace parallel
         std::map<unsigned int, pack_callback_t> pack_callbacks;
       };
 
-      CellAttachedData cell_attached_data;
+      /**
+       * Vector of tuples containing of p4est quadrant, dealii cells and their relation after refinement.
+       * This makes a one to one correlation possible and allows for parallelization in the future.
+       *
+       * To write its contents, call set_quadrant_cell_pairs(). This vector will be ordered by the occurance of
+       * quadrants in the corresponding p4est sc_array., at which data for transfer will be packed.
+       */
+      std::vector< std::tuple< typename dealii::internal::p4est::types<dim>::quadrant *,
+          typename Triangulation<dim,spacedim>::CellStatus,
+          typename Triangulation<dim,spacedim>::cell_iterator> > local_quadrant_cell_relations;
+
+      /**
+       * Go through all p4est trees and and store them in local_quadrant_cell_pairs.
+       */
+      void store_quadrant_cell_relations();
+
+      void
+      store_quadrant_cell_relations_recursively (const typename dealii::internal::p4est::types<dim>::tree &tree,
+                                                 const typename Triangulation<dim,spacedim>::cell_iterator &dealii_cell,
+                                                 const typename dealii::internal::p4est::types<dim>::quadrant &p4est_cell);
```

Comment:
same here -- here, you'd have to pass the output data structure as a non-const reference argument